recipe/patches/2667-Fix-improper-include-of-build-directory.patch

```diff
@@ -0,0 +1,35 @@
+From b77cb393b884c1dbdaffb300aa8c4967d9ba7fb0 Mon Sep 17 00:00:00 2001
+From: jhendersonHDF <jhenderson@hdfgroup.org>
+Date: Tue, 11 Apr 2023 14:31:25 -0500
+Subject: [PATCH] Fix improper include of build directory (#2422, #2621)
+ (#2667)
+
+---
+ configure.ac             |  6 ++----
+ release_docs/RELEASE.txt | 18 ++++++++++++++++++
+ 2 files changed, 20 insertions(+), 4 deletions(-)
+
+diff --git a/configure.ac b/configure.ac
+index 7c8c958bf..d5f46da75 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -3012,8 +3012,7 @@ SUBFILING_VFD=no
+ HAVE_MERCURY="no"
+
+ ## Always include subfiling directory so public header files are available
+-CPPFLAGS="$CPPFLAGS -I$ac_abs_confdir/src/H5FDsubfiling"
+-AM_CPPFLAGS="$AM_CPPFLAGS -I$ac_abs_confdir/src/H5FDsubfiling"
++H5_CPPFLAGS="$H5_CPPFLAGS -I$ac_abs_confdir/src/H5FDsubfiling"
```

Comment:
This is not enough. While it is OK for the git repo, their sources have the configure file pregenerated. I believe you need to patch the pregenerated configure file itself (not the configure.ac).

In either case (maybe I am wrong with my thoughts above), maybe you can inspect the outputs locally and ensure that the problematic variable is removed.